### ISSUE TITLE
Decorators and Babel 6

### ISSUE THREAD

**Issue by idibidiart** [wycats/javascript-decorators#46]
Any idea why Babel 6 no longer supports the decorators transform?

**Comment by loganfsmyth** [comment]
The Babel issue is tracked here: https://phabricator.babeljs.io/T2645 That's the best place for updates on Babel specifically.

As the warning says when you attempt to use decorators in Babel currently, we are waiting on adding decorators to Babel 6 currently so that we don't have to maintain two versions in core. If you look at this repo currently, it contains two very different programmatic interfaces:
1. The one in `README.md`, which is the one currently available in Babel 5, which has a big warning at the top:
   
   > This file is under active development. Refer to interop/reusability.md for the most up to date description.
2. The one in `interop/reusability.md` which is the new proposal that is currently awaiting presentation at to TC39 and the one Babel 6 will be implementing once it is ready.

**Comment by emberdart** [comment]
Everything in Babel 6 has been made into a plugin, and select sets of plugins make a preset.

See the applicable plugin here:
http://babeljs.io/docs/plugins/transform-decorators/

which is included in the stage 1 preset:
http://babeljs.io/docs/plugins/preset-stage-1/

If you have trouble, install the syntax as well:
http://babeljs.io/docs/plugins/syntax-decorators/

Cheers

**Comment by idibidiart** [comment]
@dandart 

I believe for now all that transform-decorators will do is throw an error saying decorators are not supported in Babel 6 (until TC39 committee approves the new spec)

What worked for me was using the transform-decorators-legacy plugin

**Comment by ghost** [comment]
As an an alternative to waiting to upgrade to Babel 6, or pulling out Decorators entirely, consider [restoring legacy support for Decorators](http://technologyadvice.github.io/es7-decorators-babel6/) during the upgrade.

**Comment by mattinsler** [comment]
Turning on transform-decorators-legacy will cause errors if you use class properties in a specific way.

``` javascript
class Foo {
  static SECOND = 1000;
  static MINUTE = Foo.SECOND * 60;
}
```

This will say that Foo is undefined while trying to evaluate 'Foo.SECOND'. Without the decorators-legacy plugin this works.

**Comment by loganfsmyth** [comment]
@mattinsler Up for filing that then? Commenting over here won't help fix things.

**Comment by mattinsler** [comment]
Oops, sorry, will do.

**Comment by ghost** [comment]
I'd really like to be able to use Core-Decorators and Lodash-Decorators
with Legacy! Another issue?

**Comment by loganfsmyth** [comment]
If there are issues with it, please do file them!
https://github.com/loganfsmyth/babel-plugin-transform-decorators-legacy

**Comment by ghost** [comment]
👍
